// @clonebook/models/user.js
const mongoose = require("mongoose");
const schema = mongoose.Schema;


let postSchema = schema({
    date: {type: Date, required: true, default: Date.now},
    message: {type: String, required: true},
    liked_by: []
});

let messageSchema = schema({
    message: {type: String, required: true},
    read: {type: Boolean, required: true, default: true},
    by_me: {type: Boolean, required: true, default: true},
    date: {type: Date, required: true, default: Date.now},
});

let friendSchema = schema({
    firstName: {type: String, required: true},
    lastName: {type: String, required: true},
    username: {type: String, required: true},
    status: {type: Boolean, required: true},
    avatar: {type: String, required: true},
    posts: [postSchema],
    messages: [messageSchema]
});

let invitationSchema = schema({
    firstName: {type: String, required: true},
    lastName: {type: String, required: true},
    avatar: {type: String, required: true},
});

let userSchema = schema({
    firstName:{
        type: String,
        required: true
    },
    lastName:{
        type: String,
        required: true
    },
    username:{
        type: String,
        required: true,
        unique: true
    },
    email: {
        type: String,
        required: true
    },
    password:{
        type: String,
        required: true
    },
    status:{
        type: Boolean,
        required: false,
        default: false
    },
    avatar:{
        type: String,
        required: true
    },
    posts: [postSchema],
    friends:[friendSchema],
    invitations:[invitationSchema]
});

let User = mongoose.model("User", userSchema);
module.exports = User;
